kite-data/kite-data-hive/src/main/java/org/kitesdk/data/spi/hive/HiveAbstractMetadataProvider.java

```diff
@@ -109,7 +112,25 @@ public DatasetDescriptor update(String namespace, String name, DatasetDescriptor
     String resolved = resolveNamespace(namespace, name);
     if (resolved != null) {
       Table table = getMetaStoreUtil().getTable(resolved, name);
-      HiveUtils.updateTableSchema(table, descriptor);
+
+      Path managerPath = new Path(new Path(table.getSd().getLocation()),
+          SCHEMA_DIRECTORY);
+
+      SchemaManager manager = SchemaManager.create(conf, managerPath);
+
+      DatasetDescriptor newDescriptor;
+
+      try {
+        URI schemaURI = manager.writeSchema(descriptor.getSchema());
+
+        newDescriptor = new DatasetDescriptor.Builder(descriptor)
+            .schemaUri(schemaURI).build();
```

Comment:
Yes, the underlying table has a schema URI, which is validated in one of the unit tests that creates a schema too large to fit in a literal